### PR TITLE
AP_Notify: simplify scrolling code

### PR DESCRIPTION
**Pull request by peterbarker**
Well.. I think it's simpler...

### CODE REVIEW

File: libraries/AP_Notify/Display.cpp [hunk 1]
```diff
@@ -536,40 +536,42 @@ void Display::update_text_empty(uint8_t r)
...
+    _movedelay = 0;
     _mstartpos = 0;
     draw_text(COLUMN(0), ROW(r), msg);
 }
 
 void Display::update_text(uint8_t r)
 {
-    char msg [DISPLAY_MESSAGE_SIZE] = {0};
-    char txt [NOTIFY_TEXT_BUFFER_SIZE] = {0};
+    char msg [DISPLAY_MESSAGE_SIZE] = {};
```
Comment (muramura) [inline libraries/AP_Notify/Display.cpp]: I think that it is easier to understand if you explicitly describe it.

**Comment by peterbarker** [comment]
@OXINARF Yes indeed, given these are new'd those lines could be removed!

File: libraries/AP_Notify/Display.cpp (hunk 1 above)
Comment (peterbarker) [inline libraries/AP_Notify/Display.cpp]: @muramura My reasoning here is that `{0}` is just one of many special cases you can have when initialising an array.  You could have `{0, 7, 3}` or `{2,1,4}`.

The standard just says everything that isn't nominated is zeroed, so we may as well not nominate anything.  Here we might save *two bytes*!  I'm banking those.

**Comment by peterbarker** [comment]
@OXINARF removed variable initialisation.

**Comment by OXINARF** [comment]
@peterbarker Did you test this in a LCD or is it untested? It looks good to me, but should be easy enough to test.

**Comment by peterbarker** [comment]
@OXINARF Only the ssd version.  I don't have the sh version.

**Comment by rmackay9** [comment]
I also don't have the "sh" version so i can't help with testing.
But 'cuz @OXINARF has already done all the hard work of reviewing this, I'm merging!  thanks!

**Pull request closed without merging**